plugins/askrene/layer.c

```diff
@@ -233,43 +244,45 @@ const struct local_channel *layer_find_local_channel(const struct layer *layer,
 	return local_channel_hash_get(layer->local_channels, scid);
 }
 
-static struct constraint *layer_find_constraint_nonconst(const struct layer *layer,
-							 const struct short_channel_id_dir *scidd)
+void layer_apply_constraints(const struct layer *layer,
+			     const struct short_channel_id_dir *scidd,
+			     struct amount_msat *min,
+			     struct amount_msat *max)
 {
-	return constraint_hash_get(layer->constraints, scidd);
-}
+	struct constraint *c;
+	struct constraint_hash_iter cit;
 
-/* Public one returns const */
-const struct constraint *layer_find_constraint(const struct layer *layer,
-					       const struct short_channel_id_dir *scidd)
-{
-	return layer_find_constraint_nonconst(layer, scidd);
+	/* We can have more than one: apply them all! */
+	for (c = constraint_hash_getfirst(layer->constraints, scidd, &cit);
+	     c;
+	     c = constraint_hash_getnext(layer->constraints, scidd, &cit)) {
+		if (amount_msat_greater(c->min, *min))
+			*min = c->min;
+		if (amount_msat_less(c->max, *max))
+			*max = c->max;
```

Comment:
I think this a good place to use `amount_msat_max` and `amount_msat_min`, for a tiny optimization.